src/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -30,3 +30,31 @@ Environment::ExpandEnvironmentVariables(const std::wstring & str)
 
     return expandedStr;
 }
+
+std::optional<std::wstring>
+Environment::GetEnvironmentVariableValue(const std::wstring & str)
+{
+    DWORD requestedSize = GetEnvironmentVariableW(str.c_str(), nullptr, 0);
+    if (requestedSize == 0)
+    {
+        if (GetLastError() == ERROR_ENVVAR_NOT_FOUND)
+        {
+            return std::nullopt;
+        }
+
+        throw std::system_error(GetLastError(), std::system_category(), "GetEnvironmentVariableW");
+    }
+
+    std::wstring expandedStr;
+    do
+    {
+        expandedStr.resize(requestedSize);
+        requestedSize = GetEnvironmentVariableW(str.c_str(), expandedStr.data(), requestedSize);
+        if (requestedSize == 0)
+        {
+            throw std::system_error(GetLastError(), std::system_category(), "ExpandEnvironmentStringsW");
+        }
+    } while (expandedStr.size() != requestedSize + 1);
```

Comment:
I think this while condition is right, but maybe `expandedStr.size() < requestedSize + 1` is a safer check.